### PR TITLE
chore(download): bodies downloader size limit

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Limit bodies downloader buffer based on estimated size instead of entry count.

### CODE REVIEW

File: crates/net/downloaders/src/bodies/bodies.rs [hunk 1]
```diff
@@ -223,21 +223,24 @@ where
     fn pop_buffered_response(&mut self) -> Option<OrderedBodiesResponse> {
         let resp = self.buffered_responses.pop()?;
         self.metrics.buffered_responses.decrement(1.);
-        self.num_buffered_blocks -= resp.len();
-        self.metrics.buffered_blocks.set(self.num_buffered_blocks as f64);
-        self.metrics.buffered_blocks_size_bytes.decrement(resp.size() as f64);
+        self.buffered_blocks_size_bytes -= resp.size();
+        self.metrics.buffered_blocks.decrement(resp.len() as f64);
```
Comment (mattsse) [inline crates/net/downloaders/src/bodies/bodies.rs:227]: this can now use `.set(` as well

**Review by mattsse — APPROVED** [review]
lgtm,

smol nit, 4GB seems reasonable

File: crates/net/downloaders/src/bodies/bodies.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/net/downloaders/src/bodies/bodies.rs:227]: i removed `num_buffered_blocks` tho

Comment (mattsse) [reply crates/net/downloaders/src/bodies/bodies.rs:227]: ah I see

**Pull request merged**